### PR TITLE
Initial commit from poc with changes compatible with OSD 3.0

### PR DESCRIPTION
**Pull request by Shivamdhar**
Signed-off-by: Shivam Dhar <dhshivam@amazon.com>

### Description

Phase 1: Basic framework for maps project
  - Adding map layers POC to dashboards-maps
  -  Ensuring compatibility with OSD 3.0
 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/plugins/maps-dashboards-plugin/public/common/opensearch_maps_client.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    this._manifestServiceUrl = manifestServiceUrl;
+  }
+
+  async isEnabled() {
+    let result;
+    try {
+      result = await this._fetchWithTimeout(this._manifestServiceUrl);
+    } catch (e) {
+      // silently ignoring the exception and returning false.
+      return false;
```
Comment (junqiu-lei) [inline src/plugins/maps-dashboards-plugin/public/common/opensearch_maps_client.js:24]: Based on last bug fix on dashboards repo, here we want to return true. https://github.com/opensearch-project/OpenSearch-Dashboards/pull/1550/files#diff-30df0a18dfe7348e10bec7d9782208a172c77771c113061b47de51a25b22f2cd

File: src/plugins/maps-dashboards-plugin/public/map/service_settings.js [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+        tileApiUrl: this._mapConfig.emsTileApiUrl,
+        landingPageUrl: this._mapConfig.emsLandingPageUrl,
+        fetchFunction: function (...args) {
+          return fetch(...args);
+        },
+      });
+    }
+  }
+
+  async getFileLayers() {
```
Comment (junqiu-lei) [inline src/plugins/maps-dashboards-plugin/public/map/service_settings.js:152]: same here, we will need to use the latest bug fix code. https://github.com/opensearch-project/OpenSearch-Dashboards/pull/1550/files#diff-edd35a21ee8bbae0d475be1ab224deda046e45c13d0d09b752c9b43b7da6a2ff

**Comment by VijayanB** [comment]
Why are we merging this way instead of building bottom up based on feature?

**Comment by Shivamdhar** [comment]
We won't be ingesting maps-explorer code at the moment.

**Pull request closed without merging**